src/types.d.ts

```diff
@@ -1,4 +1,4 @@
-interface IndexedCollection {
+interface IndexedCollection extends Iterable<number> {
```

Comment:
What's the use case for making it iterable?

Reply:
Without it passing mat4/vec4 objects to `WebGLRenderingContext.uniformMatrix[234]fv()` fails. See [here](https://github.com/microsoft/TypeScript/blob/deb5bac520a1134e188fc6f01fa4f52b66d2899b/lib/lib.dom.iterable.d.ts#L332-L340).

Reply:
Thank you for thoroughly testing everything!